packages/math-input/src/consts.js

```diff
@@ -1,73 +1,74 @@
+// @flow
 /**
  * Constants that are shared between multiple files.
  */
 
-module.exports = {
-    KeypadTypes: {
-        FRACTION: "FRACTION",
-        EXPRESSION: "EXPRESSION",
-    },
+export const KeypadTypes = {
+    FRACTION: "FRACTION",
+    EXPRESSION: "EXPRESSION",
+};
 
-    KeyTypes: {
-        EMPTY: "EMPTY",
-        // For numerals, variables, and any other characters that themselves
-        // compose 'values'.
-        VALUE: "VALUE",
-        // For buttons that insert or adjust math in an input.
-        OPERATOR: "OPERATOR",
-        // For buttons that move the cursor in an input (including via
-        // deletion).
-        INPUT_NAVIGATION: "INPUT_NAVIGATION",
-        // For buttons that modify the broader keypad state (e.g., by changing
-        // the visible pane).
-        KEYPAD_NAVIGATION: "KEYPAD_NAVIGATION",
-        // For buttons that house multiple buttons and have no action
-        // themselves.
-        MANY: "MANY",
-        // For the echo animation that appears on press.
-        ECHO: "ECHO",
-    },
+export const KeyTypes = {
+    EMPTY: "EMPTY",
+    // For numerals, variables, and any other characters that themselves
+    // compose 'values'.
+    VALUE: "VALUE",
+    // For buttons that insert or adjust math in an input.
+    OPERATOR: "OPERATOR",
+    // For buttons that move the cursor in an input (including via
+    // deletion).
+    INPUT_NAVIGATION: "INPUT_NAVIGATION",
+    // For buttons that modify the broader keypad state (e.g., by changing
+    // the visible pane).
+    KEYPAD_NAVIGATION: "KEYPAD_NAVIGATION",
+    // For buttons that house multiple buttons and have no action
+    // themselves.
+    MANY: "MANY",
+    // For the echo animation that appears on press.
+    ECHO: "ECHO",
+};
 
-    DeviceOrientations: {
-        LANDSCAPE: "LANDSCAPE",
-        PORTRAIT: "PORTRAIT",
-    },
+export const DeviceOrientations = {
+    LANDSCAPE: "LANDSCAPE",
+    PORTRAIT: "PORTRAIT",
+};
 
-    DeviceTypes: {
-        PHONE: "PHONE",
-        TABLET: "TABLET",
-    },
+export const DeviceTypes = {
+    PHONE: "PHONE",
+    TABLET: "TABLET",
+};
 
-    LayoutModes: {
-        FULLSCREEN: "FULLSCREEN",
-        COMPACT: "COMPACT",
-    },
+export const LayoutModes = {
+    FULLSCREEN: "FULLSCREEN",
+    COMPACT: "COMPACT",
+};
 
-    BorderDirections: {
-        LEFT: "LEFT",
-        BOTTOM: "BOTTOM",
-    },
-    BorderStyles: {
-        LEFT: ["LEFT"],
-        BOTTOM: ["BOTTOM"],
-        ALL: ["LEFT", "BOTTOM"],
-        NONE: [],
-    },
+export const BorderDirections = {
+    LEFT: "LEFT",
+    BOTTOM: "BOTTOM",
+};
+export const BorderStyles: {|
+    [style: string]: $ReadOnlyArray<$Values<typeof BorderDirections>>,
+|} = {
```

Comment:
This unfortunately doesn't do what you think it does.  The problem is that the `"LEFT"` and `"BOTTOM"` are inferred as `string`s.  In order for Flow to treat string literals as the literal type and not `string` you'd have to type `BorderDirections` in the following way:
```
export const BorderDirections {
    LEFT: ("LEFT": "LEFT"),
    BOTTOM: ("BOTTOM": "BOTTOM"),
};
```
I don't think this is important right now, but it might be good to leave a comment for whenever y'all get around to enabling Flow in math-input.

Reply:
Thanks for thinking of this! 

I think in the future I'd like to move to either use a simple type union (`export const BorderDirections = "LEFT" | "BOTTOM";`) or true Flow enums. These being objects feels unnecessary in Flow.

Reply:
My vote would be for actual enums.  It makes updating the code easier if you want to change something.